File: data/compilers_comparison/src/examples/second_example/src/func17.c
#include "second_example.h" 
array_t* func17(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array168;
   if (pCounter > 0) {
      array168 = vars->data[--pCounter];
      array168->refC++;
      DEBUG_COPY(array168->id);
   } else {
      array168 = (array_t*)malloc(sizeof(array_t));
      array168->size = 80;
      array168->refC = 1;
      array168->id = 168;
      array168->data = (unsigned int*)malloc(array168->size*sizeof(unsigned int));
      memset(array168->data, 0, array168->size*sizeof(unsigned int));
      DEBUG_NEW(array168->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array168;
   array_t* array169 = func24(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array169->id);
   free(params0.data);
   unsigned int loop49 = 0;
   unsigned int loopLimit49 = (rand()%loopsFactor)/3 + 1;
   for(; loop49 < loopLimit49; loop49++) {
      for (int i = 0; i < array169->size; i++) {
         array169->data[i]++;
      }
   }
   for (int i = 0; i < array169->size; i++) {
      if (array169->data[i] == 12) { 
         return array169;
      }
   }
   array169->refC--;
   if(array169->refC == 0) {
      free(array169->data);
      free(array169);
      DEBUG_FREE(array169->id);
   }
   return array168;
}

File: data/multilang_comparison/program_nim/src/func17.nim
proc func17*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array200: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array200 = vars.data[pCounter]
      inc array200.refC
      debugCopy(array200.id)
   else:
      array200 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array200.size = 51
      array200.refC = 1
      array200.id = 200
      array200.data = cast[ptr UncheckedArray[uint32]](alloc(array200.size * sizeof(uint32)))
      zeroMem(array200.data, array200.size * sizeof(uint32))
      debugNew(array200.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array200
   var array201: ptr ArrayT = func24(addr params0, getPath(), loopsFactor)
   debugReturn(array201.id)
   dealloc(params0.data)
   let loopLimit58 = (50) div (3) + 1
   for loop58 in 0 ..< loopLimit58:
      discard
      for i in 0 ..< array201.size:
         inc array201.data[i]
   for i in 0 ..< array201.size:
      if array201.data[i] == 27:
         return array201
   dec array201.refC
   if array201.refC == 0:
      dealloc(array201.data)
      dealloc(array201)
      debugFree(array201.id)
   return array200

